### PR TITLE
Add EBS gp3 support

### PR DESCRIPTION
**Pull request by samjo-nyang**
**Is this a bug fix or adding new feature?**
Adding new feature

**What is this PR about? / Why do we need it?**
Support a new volume type gp3 (https://aws.amazon.com/about-aws/whats-new/2020/12/introducing-new-amazon-ebs-general-purpose-volumes-gp3/)

**What testing is done?** 
- `make` succeeded
- `make sanity-test` succeeded
- manual gp3 pvc testing (with my internal EKS cluster) succeeded

implements #628

However, `docker run -v "$PWD":/go -it golang:1.14.1-buster make test` triggers a lot of dependency update. Is this expected and should be in this PR? How can I update aws-sdk only?

### CODE REVIEW

**Comment by samjo-nyang** [comment]
I signed it

**Comment by ayberk** [comment]
This is great, thank you!

File: pkg/cloud/cloud.go
Comment (Jan-M) [inline pkg/cloud/cloud.go:262]: From my point of view I would like no connection seen between volume size and IOPS. gp3 allows you to spec any any value for IOPS once volume exceeds 32gb. It gives end user 3000IOPS base line, disconnected from the size.

Comment (samjo-nyang) [reply pkg/cloud/cloud.go:262]: I also agree on this comment; should I
1. give another option for gp3 only such as "iopsTotal"?
2. give the "iopsTotal" option and migrate io1 and io2 iopsPerGB to iopsTotal?
3. just simply do max(3000, capacityGB * iopsPerGB)?

Comment (ayberk) [reply pkg/cloud/cloud.go:262]: +1. This is the biggest selling point of gp3. You don't need to set iops while creating a gp3 volume(see [doc](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVolume.html)], so you can simply delete this branch and add `VolumeTypeGP3` to the first case block. Line 308 below will set the request.Iops if it's passed by the user.

**Review by ayberk — CHANGES_REQUESTED** [review]
Again, great work. Thank you!

We should probably change the default type to gp3 as it's cheaper and faster. What do you think @wongma7?

**Comment by wongma7** [comment]
/ok-to-test

IMO we can make gp3 the default if we release the change in a new minor version, by default the performance is equal or better, only throughput is a bit more complicated since for gp2 it might involve "burst credits" but I don't think we need to worry about it. Also there are volume type specific service quotas, so for example if a user has increased their gp2 TiB quota, they must remember to increase their gp3 one as well, but again I don't think this is worth worrying about just somethin to note.

File: pkg/cloud/cloud.go
Comment (Jan-M) [reply pkg/cloud/cloud.go:262]: But the first case uses the odd `iopsPerGB` which someone familiar with backwards compatibility would need to judge changing.

So having the new case allows changing to plain iops and throughput.

File: pkg/cloud/cloud.go
Comment (ayberk) [inline pkg/cloud/cloud.go:311]: Can you change this to `throughput > 0 && diskOptions.VolumeType == VolumeTypeGP3`? Let's play it safe as throughput is a valid parameter only for gp3.

File: pkg/cloud/cloud.go
Comment (ayberk) [reply pkg/cloud/cloud.go:262]: You're right! We need to add another field "iops" so the users can directly set the value.

File: docs/README.md
Comment (ayberk) [inline docs/README.md:36]: `and optional when gp3 volume type is specified` this doesn't apply anymore.

File: pkg/cloud/cloud.go
Comment (ayberk) [inline pkg/cloud/cloud.go:144]: I think we should follow the API's naming and call this "iops".

File: tests/e2e/driver/ebs_csi_driver.go
Comment (ayberk) [inline tests/e2e/driver/ebs_csi_driver.go:113]: I see other fields are not using them, but can you use the constants you defined above? Ie `IopsTotalKey` and `ThroughputKey `

**Comment by samjo-nyang** [comment]
I think that all feedback are applied and all tests are passed.
- change default volume type to gp3
- add a new option "iops"
- fix readme.md

@ayberk @wongma7 @Jan-M 
Is there anything left to do?

**Comment by ayberk** [comment]
Thank you @samjo-nyang! /lgtm, but let's wait for one more review. 

One minor request, can you squash your commits? Ideally we should have one commit for updating the SDK and another one for gp3 support, but you can simply squash into one.

**Comment by wongma7** [comment]
/lgtm
/approve
thank you, we will make a release shortly!!

**Pull request merged**
